src/Search/Api/Elasticsearch/ElasticsearchClient.php

```diff
@@ -28,9 +34,22 @@ public function deleteIndexByName($index)
         return $this->connection->indices()->delete($params);
     }
 
-    public function deleteIndex()
+    public function createIndex($indexName){
+
+        $params = [
+            'index' => $indexName
+
+        ];
+        return $this->connection->indices()->create($params);
+
+    }
+
+    public function deleteIndex($indexName ='')
     {
-        return $this->deleteIndexByName($this->index);
+        if (empty($indexName)){
+            $indexName = $this->index;
+        }
+        return $this->deleteIndexByName($indexName);
```

Comment:
If the default is null, this could be simplified to `$indexName ?? $this->index` (applied to other methods too)